modules/siddhi-core/src/main/java/io/siddhi/core/query/SelectStoreQueryRuntime.java

```diff
@@ -71,7 +74,27 @@ public SelectStoreQueryRuntime(QueryableProcessor queryableProcessor, CompiledCo
 
     @Override
     public void reset() {
+        if (selector != null) {
+            selector.process(generateResetComplexEventChunk(metaStreamEvent));
+        }
+    }
+
+    private ComplexEventChunk<ComplexEvent> generateResetComplexEventChunk(MetaStreamEvent metaStreamEvent) {
+        StreamEvent streamEvent = new StreamEvent(metaStreamEvent.getOutputData().size(),
```

Comment:
Can't we create an event factory and use that, why are we creating events every time from the MetaStreamEvent?

Reply:
this is how it is implemented in all store query runtimes https://github.com/niruhan/siddhi/blob/53466a751c2bea850aa9fed867fd9e61d903f2c9/modules/siddhi-core/src/main/java/io/siddhi/core/query/FindStoreQueryRuntime.java#L134

Reply:
ok